### DISCUSSION TITLE
Lower the Minimum requirement to Wrapeth

### DISCUSSION THREAD

**Discussion by wainesiro** [ProjectOpenSea/opensea-js#1469] [Q&A]
Hello! when i try to wrap eth less the 1eth via sdk. i keep getting errors that amount is too low! can we lower the minimum requirement?

**Comment by ryanio** [comment]
There is no minimum, you might be misunderstanding the error message, can you share it here?

**Reply by wainesiro** [reply]
"Wraping failed error (attempt 2): TypeError: underflow (argument="value", value=0.2, code=INVALID_ARGUMENT, version=6.12.0)"
i am getting this error when i try to wrap eth less then "1" but it works perfectly when i try to wrap more then 1
"it maybe because Wrapeth in sdk only accept bignumberish"

**Comment by wainesiro** [comment]
"Wraping failed error (attempt 2): TypeError: underflow (argument="value", value=0.2, code=INVALID_ARGUMENT, version=6.12.0)"
i am getting this error when i try to wrap eth less then "1" but it works perfectly when i try to wrap more then 1
"it maybe because Wrapeth in sdk only accept bignumberish" @ryanio

**Comment by ryanio** [comment]
Ah thanks for the error, need to convert it from ether to wei, I will work on a PR to fix this.

**Comment by ryanio** [comment]
What OpenSea-js version are you using? Can you try with the latest version? I see in the wrapETH code we should be converting it to the wei value

**Reply by wainesiro** [reply]
i am using the latest version only but still its the same

**Reply by ryanio** [reply]
can you share your problematic code? i cannot help further without seeing that

**Reply by wainesiro** [reply]
> can you share your problematic code? i cannot help further without seeing that

"async function Wrapeth(amountInEth, accountAddress) {
  const maxRetries = 3; // Define the maximum number of retries

  async function WrappingWithRetry() {
    for (let retry = 0; retry < maxRetries; retry++) {
      try {
        const order = await openseaSDK.wrapEth({
          amountInEth: amountInEth,
          accountAddress: accountAddress,
        });
        return order;
      } catch (error) {
        console.error(`Wraping failed error (attempt ${retry + 1}):`, error);
        await sleep(5000); // Wait for 5 seconds before retrying
      }
    }
    return null; // If max retries reached, return null or handle as needed
  }

  return WrappingWithRetry();
}"

I am using the following code and it works perfectly when i wrap more then "1" but gives error when its less then "1"
